test: rewrite cutler to be testable

eseidel opened the pull request:
I removed the 'rebase' command for now since it was dead.
'rebase' should be re-written later to just print the commands
needed since having it actually drive the rebase was a dream.
I separated "Repo" from "Checkout" and removed the global config object.
I moved to using a global logger and package scoped to match
other packages.
I added a bunch of tests.  This still doesn't have 100% coverage
but it should
be 100% testable now.

eseidel committed:
attempt to fix cutler

eseidel committed:
feat: improve cutler output

I also shared some code between "versions" and "rebase"

bryanoltman commented on packages/cutler/lib/git_extensions.dart:
```diff
@@ -50,20 +50,35 @@ extension RepoCommands on Repo {
 
   /// Returns a [Version] for the given [hash].
-  Version versionFrom(String hash, {bool lookupTags = true}) {
+  Version versionFrom(String commitish, {bool lookupTags = true}) {
```
> commitish

Typo?

eseidel replied:
https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefcommit-ishacommit-ishalsocommittish

bryanoltman replied:
Huh! TIL

bryanoltman replied:
Probably should update the doc comment to reflect the argument rename

eseidel replied:
Fixed.

eseidel committed:
test: re-write cutler to be more testable

I removed the 'rebase' command for now since it was dead.
'rebase' should be re-written later to just print the commands
needed since having it actually drive the rebase was a dream.
I separated "Repo" from "Checkout" and removed the global config object.
I moved to using a global logger and package scoped to match
other packages.
I added a bunch of tests.  This still doesn't have 100% coverage
but it should
be 100% testable now.

eseidel committed:
fix commitish in arg docs

eseidel commented:
I ended up doing a massive rewrite of cutler.

eseidel committed:
Set user.email and user.name
